_search-plugins/search-pipelines/ml-inference-search-response.md

```diff
@@ -388,4 +388,368 @@ The response confirms that the processor has generated text embeddings in the `p
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "dataAsMap": {
+            "label": "LABEL_0",
+            "score": 0.022704314440488815
+          }
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+**Step 1: Index sample documents**
+
+Create an index and add some sample documents to it:
+
+```json
+POST _bulk
+{"index":{"_index":"demo-index-0","_id":"1"}}
+{"diary":"I hate you"}
+{"index":{"_index":"demo-index-0","_id":"2"}}
+{"diary":"I love you"}
+{"index":{"_index":"demo-index-0","_id":"3"}}
+{"diary":"I dislike you"}
+```
+{% include copy-curl.html %}
+
+**Step 2: Create a search pipeline**
+
+For this example, you'll create a search pipeline that uses a text similarity model in a `one-to-one` inference mode, processing each document in the search results individually. This setup allows the model to make one prediction request per document, providing specific relevance insights for each search hit. When using `input_map` to map the search request to query text, the JSON path must start with `$._request` or `_request`:
+
+```json
+PUT /_search/pipeline/my_rerank_pipeline
+{
+  "response_processors": [
+    {
+      "ml_inference": {
+        "tag": "ml_inference",
+        "description": "This processor runs ml inference during search response",
+        "model_id": "Ialx65IBAs32TwoK1lXf",
+        "model_input":"""{"parameters":{"inputs":{"text":"${input_map.text}","text_pair":"${input_map.text_pair}"}}}""",
+        "function_name": "REMOTE",
+        "input_map": [
+          {
+            "text": "diary",
+            "text_pair":"$._request.query.term.diary.value"
+          }
+        ],
+        "output_map": [
+          {
+            "rank_score": "$.score"
+          }
+        ],
+        "full_response_path": false,
+        "model_config": {},
+        "ignore_missing": false,
+        "ignore_failure": false,
+        "one_to_one": true
+        },
+        "rerank": {
+          "by_field": {
+            "target_field": "rank_score",
+            "remove_target_field": true
+          }
+        }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+In this configuration, you're providing the following parameters:
+
+- The `model_id` specifies the unique identifier of the text similarity model.
+- The `function_name` is set to `REMOTE`, indicating that the model is hosted externally.
+- The `input_map` maps the `diary` field from each document to the `text` input of the model, and the search query term to the `text_pair` input.
+- The `output_map`  maps the model's score to a field named `rank_score` in each document.
+- The `model_input` formats the input for the model, ensuring it matches the structure expected by the Predict API.
+- The `one_to_one` parameter is set to `true`, ensuring that the model processes each document individually, rather than batching multiple documents together.
```

Comment:
```suggestion
- The `one_to_one` parameter is set to `true`, ensuring that the model processes each document individually rather than batching multiple documents together.
```